tests/kernel/workq/work_queue/src/start_stop.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024 Måns Ansgariusson <mansgariusson@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+#define NUM_TEST_ROUNDS 5
+#define NUM_TEST_ITEMS  10
+/* Each work item takes 100ms by default. */
+#define WORK_ITEM_WAIT  (CONFIG_TEST_WORK_ITEM_WAIT_MS)
+/* In fact, each work item could take up to this value */
+#define WORK_ITEM_WAIT_ALIGNED                                                                     \
+	k_ticks_to_ms_floor64(k_ms_to_ticks_ceil32(WORK_ITEM_WAIT) + _TICK_ALIGN)
+#define SUBMIT_WAIT (CONFIG_TEST_SUBMIT_WAIT_MS)
+#define STACK_SIZE  (1024 + CONFIG_TEST_EXTRA_STACK_SIZE)
+#define CHECK_WAIT  ((NUM_TEST_ITEMS + 1) * WORK_ITEM_WAIT_ALIGNED)
+
+static struct k_work_q work_q;
+static K_THREAD_STACK_DEFINE(work_q_stack, STACK_SIZE);
+
+struct k_work works[NUM_TEST_ITEMS];
+
+static atomic_t num_work_executed;
+static void work_handler(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	atomic_inc(&num_work_executed);
+}
```

Comment:
It probably isn't necessary to use atomics here (or even have a counter).

I would suggest adding a simple 100ms sleep inside of the work handler.

That should be sufficient time to verify that the work queue is busy in the main test thread.